### DISCUSSION TITLE
Send batch transactions on ZkSync

### DISCUSSION THREAD

**Discussion by seddik11** [zkSync-Community-Hub/zksync-developers#146] [General] [answered]
### Environment

Testnet

### Provide a brief description of the functionality you're trying to implement.

Hi, is there a way to send a batch of signed transaction on ZkSync ? the goal here is to reduce the gas fees.
Is there some sdk ? or some method using smart accounts (like call a function with an array of transactions) ?

### What is the specific issue or error you're encountering?

No

### Can you share the error messages or logs you're receiving, if any?

No

### Have you made any recent changes to the contract before encountering this issue?

No

### Are there any external libraries or contracts that your contract interacts with?

No

### Can you provide the relevant portions of your contract code where the issue is occurring?

No

### Have you tried to isolate the problem, and if so, what were the results?

No

### What steps have you already taken to try to resolve the issue?

No

**Comment by mylastwanto** [comment]
Have you try multisender.app? The contract code can easily be seen on Etherscan

After a quick look, I'm sure the code can be used on zksync era

**Reply by seddik11** [reply]
It's just a service for transfering erc20 or nfts token. I would like to make multiple signed transactions signed by different users (msg.sender) into the same batch to reduce gas fees. It's like the layer 3 idea which will make gas cheaper

**Comment by idea404** [comment, marked as answer]
If you're looking to batch transactions using a multi-call contract on zkSync Era, here are some resources and addresses that might be helpful:

Multi-call Contract Repository: [GitHub link](https://github.com/mds1/multicall)

Deployment to zkSync Era: [Deployment details](https://github.com/mds1/multicall/blob/b5026d38a12378932a85ba3237b4a1b1a1dacb09/deployments.json#L564)

Another useful resource: [GitHub link](https://github.com/Destiner/ethcall/blob/master/src/multicall.ts)

**Deployed Addresses**:

**Testnet**:
Multicall1 address: `0x52806DC8c5Cb95a560039409FE2e1C29d9eBb22f`
Multicall2 address: `0xDED0Aa78681A5AbF85D830D4478cf4acF00Ec4a4`
Multicall3 address: `0x413fEb613604D46586c22801949A5b88b224c260`

**Mainnet**:
Multicall1 address: `0x52806DC8c5Cb95a560039409FE2e1C29d9eBb22f`
Multicall2 address: `0xDED0Aa78681A5AbF85D830D4478cf4acF00Ec4a4`
Multicall3 address: `0x413fEb613604D46586c22801949A5b88b224c260`

Let me know if you have any other questions or need further assistance!

**Comment by inesarmetessa** [comment]
To send a batch of signed transactions on zkSync, you can use the zkSync JavaScript SDK. It provides methods for batching transactions, which can help reduce gas fees by combining multiple transactions into a single batch. You can find detailed documentation and examples on how to use the SDK on the zkSync developer portal.

**Reply by seddik11** [reply]
could you provide me the link for this SDK and the right API ?

**Comment by davenportbri** [comment]
Yes, you can send a batch of signed transactions on zkSync to reduce gas fees. zkSync provides support for batch transactions through its SDKs. You can use the SDKs provided by zkSync to create and send batch transactions efficiently. Make sure to follow the documentation and examples provided by zkSync for implementing batch transactions effectively.

**Reply by seddik11** [reply]
could you provide me the link for this SDK and the right API ?

**Comment by eleykelly** [comment]
Feature Request: Implement batch transaction functionality in ZkSync to reduce gas fees.
Potential Solutions: Explore SDKs or smart contract methods for sending batches of signed transactions.

**Comment by latoyatidwell** [comment]
Yes, you can send a batch of signed transactions on ZkSync to reduce gas fees. ZkSync provides SDKs for various programming languages such as JavaScript and Rust, which allow you to create and send batches of transactions efficiently. Additionally, you can use smart contracts to execute multiple transactions atomically in a single function call, further optimizing gas usage. Make sure to explore the ZkSync documentation and SDKs for more detailed information and examples on batch transactions.

**Comment by ngopriyanka** [comment]
Yes, it is possible to send a batch of signed transactions on ZkSync to reduce gas fees. The zksync SDK provides a method called BatchBuilder that allows you to construct and sign a batch of transactions before submitting them to the network. This can significantly reduce gas costs compared to submitting each transaction individually. Alternatively, you can use smart contracts to batch transactions by creating a function that accepts an array of transactions and executes them all at once.
